aws/resource_aws_elasticsearch_domain.go

```diff
@@ -548,7 +548,7 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 	}
 	if ds.SnapshotOptions != nil {
```

Comment:
This `d.Set()` was incorrectly implemented before (notice the missing `[]` before the map) and we should probably fix it now. I think the AutomatedSnapshotStartHour value also needs to be wrapped in `int()`. Something like the below could handle both for now and return an error if incorrect:

```go
if err := d.Set("snapshot_options", flattenESSnapshotOptions(ds.SnapshotOptions)); err != nil {
	return fmt.Errorf("error setting snapshot_options": err)
}
```

```go
func flattenESSnapshotOptions(snapshotOptions *elasticsearch.SnapshotOptions) []map[string]interface{} {
	if snapshotOptions == nil {
		return []map[string]interface{}{}
	}

	m := map[string]interface{}{
		"automated_snapshot_start_hour": int(aws.Int64Value(snapshotOptions.AutomatedSnapshotStartHour)),
	}

	return []map[string]interface{}{m}
}
```